### PR TITLE
SDI-357 add migrate script

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/controllers/RestrictedPatientsBatchController.kt [hunk 1]
```diff
@@ -39,6 +39,6 @@ class RestrictedPatientsBatchController(
   /**
    * DRY RUN version - this will validate input only but perform no actions
    */
-  fun dryRunProcessUnknownPatients(@RequestBody patients: List<String>): List<UnknownPatientResult> =
-    unknownPatientService.migrateInUnknownPatientsDryRun(patients)
+  fun dryRunProcessUnknownPatients(@RequestBody patient: String): UnknownPatientResult =
+    unknownPatientService.migrateInUnknownPatient(patient, true)
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/controllers/RestrictedPatientsBatchController.kt:43]: Decided to perform the dry run on individual patients too as this makes the migration script much simpler.

File: scripts/migrate-unknown.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+#set -x
+
+TOKEN=none
+ENV=dev
+DRY_RUN=1
+
+while [[ $# -gt 0 ]]; do
```
Comment (petergphillips) [inline scripts/migrate-unknown.sh:9]: I'd normally use `getopts` instead e.g. https://github.com/ministryofjustice/hmpps-auth/blob/main/rotate_clientID_cloudplatform_app.sh#L21

File: scripts/migrate-unknown.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      shift # past argument
+      shift # past value
+      ;;
+    -e)
+      ENV=$2
+      shift # past argument
+      shift # past value
+      ;;
+    --no-dry-run)
+      DRY_RUN=0
```
Comment (petergphillips) [inline scripts/migrate-unknown.sh:22]: might be better to just set `REQ_PATH` here rather than having separate variable

**Review by petergphillips — APPROVED** [review]
LGTM

File: scripts/migrate-unknown.sh [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    continue
+  fi
+
+  RESULT=$(curl -sS --http1.1 -X POST "https://restricted-patients-api$ENV_SUFFIX.hmpps.service.justice.gov.uk/$REQ_PATH" \
+    -H "Authorization: Bearer ${TOKEN}" \
+    -H "Content-Type: application/json" \
+    -d "$unknownPatient")
+  STATUS="$?"
+
+  if [ "$STATUS" -eq 0 ]
```
Comment (petergphillips) [inline scripts/migrate-unknown.sh:59]: normally better to do `if [[ ]]` instead

**Pull request merged**